From 8fe6332b0ef42d46509373fc158c266de6e4d45d Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Wed, 14 Nov 2018 18:52:33 +0800
Subject: [PATCH] LocaleTracker: Add null check before accessing WifiManager

 * WiFi service is not considered as a core service so it won't start in
   min framework, but LocaleTracker accesses to WiFi service's API.
   This would lead to a fatal exception and trigger RescueParty.
 * Add null check before accessing WifiManager to avoid the crash.
 * Remove useless casting while at it.

Change-Id: Ib1e9aef5a34cb8cfc33f7aa6f242807bc6ef2c46
Signed-off-by: Satish Patel <tosatishpatel@gmail.com>
---
 src/java/com/android/internal/telephony/LocaleTracker.java | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/internal/telephony/LocaleTracker.java b/src/java/com/android/internal/telephony/LocaleTracker.java
index 996e288..bf7c98b 100644
--- a/src/java/com/android/internal/telephony/LocaleTracker.java
+++ b/src/java/com/android/internal/telephony/LocaleTracker.java
@@ -408,8 +408,11 @@ public class LocaleTracker extends Handler {
             // Set the country code for wifi. This sets allowed wifi channels based on the
             // country of the carrier we see. If we can't see any, reset to 0 so we don't
             // broadcast on forbidden channels.
-            ((WifiManager) mPhone.getContext().getSystemService(Context.WIFI_SERVICE))
-                    .setCountryCode(countryIso);
+            WifiManager wifiManager = mPhone.getContext().getSystemService(
+                    WifiManager.class);
+            if (wifiManager != null) {
+                wifiManager.setCountryCode(countryIso);
+            }
         }
     }
 
-- 
2.7.4

